lib/ocamlorg_web/graphql.mli

```diff
@@ -5,41 +5,85 @@ type package_info =
   ; constraints : string option
   }
 
-type packages_result =
+type packages_success =
   { total_packages : int
   ; packages : Package.t list
   }
 
-val starts_with : string -> string -> bool
-(** [starts_with] function compares two strings and returns true if the first
-    string is equal to the beginning of the second string up to capitalization *)
+type packages_result = (packages_success, [ `Msg of string ]) result
 
-val is_package : string -> string -> bool
-(** [is_package] function compares two strings and returns true if they are
-    equal up to capitalization *)
+type package_success = { package : Package.t }
 
-val get_packages_result
-  :  int
+type package_result = (package_success, [ `Msg of string ]) result
+
+type params_validity =
+  | Valid_params
+  | Wrong_limit
+  | Wrong_offset
+
+val get_info : (Package.Name.t * string option) list -> package_info list
+(** [get_info] function loops through the list part of Package.Info such as
+    Package.Info.dependencies and returns a list of name and constraints record
+    Package.Info record *)
+
+(* val is_in_range : 'a -> 'a -> 'a -> bool *)
+val is_in_range : 'a -> 'a option -> 'a option -> bool
```

Comment:
@patricoferris please do you mean something like this: 
```
   let is_in_range current_version from_version upto_version =
  let current_version = Package.Version.to_string current_version in
  match from_version, upto_version with
  | None, None ->
    true
  | Some from_version, None ->
    current_version >= (Package.Version.to_string from_version)
  | None, Some upto_version ->
    current_version <= (Package.Version.to_string upto_version)
  | Some from_version, Some upto_version ->
   current_version >= (Package.Version.to_string from_version) && current_version <= (Package.Version.to_string upto_version)
```
Or  you are suggesting we use `OpamVersion.compare v1 v2` to compare the versions

Reply:
The latter, use `OpamVersion.compare` but here it would be `Package.Version.compare` (they're the same function) see: https://github.com/ocaml/v3.ocaml.org-server/blob/d7899a5476555c24c97255d4dc784c07af795a0a/lib/ocamlorg/package.mli#L17-L26

Then you can update the `mli` to be something like :
```ocaml
val is_in_range : Package.Version.t -> Package.Version.t option -> Package.Version.t option -> bool
```

Alternatively, you could push this function back into the `lib/ocamlorg/package.ml` so others can use it too if that makes more sense. Either is fine by me :))

Reply:
I think `val compare : t -> t -> int` returns an int but we need a bool for the filter function

Reply:
Yes you will need to use the `int` it returns to reconstruct the logic you already have. For example consider this case:

```ocaml
| Some from_version, None ->
    current_version >= from_version
```

Would become `Package.Version.compare current_version from_version >= 0`. From the Stdlib docs on compare functions:

> compare x y returns 0 if x is equal to y, a negative integer if x is less than y, and a positive integer if x is greater than y.